docs/public/deploying-java-app-with-database.adoc

```diff
@@ -0,0 +1,432 @@
+= Binding a Java microservices JPA application to an in-cluster Operator-Managed PostgreSQL Database on minikube
+
+This scenario illustrates binding an `odo` managed Java MicroServices JPA application to an in-cluster Operator managed PostgreSQL Database in the minikube environment.
+
+== Installation and Configuration of the minikube environment
+[NOTE]
+====
+The guide only works with minikube configured with kubernetes 1.19.x or lower. `odo link` cannot link services successfully in a Kubernetes 1.20.x or newer environment.
+====
+
+It is recommended that users of this guide obtain a suitable system for running minikube with kubernetes. In practice this should be a 4 core system minimum. Before proceeding to the sample application, please follow the instructions for establishing a minikube environment:
+
+=== Installing and starting docker 
+To install docker follow link:https://docs.docker.com/engine/install/[the Docker documentation].
+
+=== Installing, starting and configuring minikube
+
+==== Installing minikube
+To install minikube follow link:https://minikube.sigs.k8s.io/docs/start/[minikube installation instructions] for your operating system.
+
+==== Starting minikube
+
+* If you are trying to run minikube as a root user, minikube throws an error that docker should not be run as root as a matter of practice and aborts the start up. To proceed, start minikube in a manner which will override this protection mechanism:
++
+[source,sh]
+----
+$ minikube start --force --driver=docker --kubernetes-version=v1.19.8
+----
+
+* If you are a non-root user, start minikube as usual:
++
+[source,sh]
+----
+$ minikube start --kubernetes-version=v1.19.8
+----
+
+==== Configuring minikube
+The application requires an ingress add-on to allow the routes to be created easily. Configure minikube for ingress by adding ingress as a minikube add-on:
+[source,sh]
+----
+> minikube addons enable ingress
+----
+
+It is possible that you face the Dockerhub pull rate limit if you do not have a pull secret for your personal free Dockerhub account. During ingress initialization two of the job pods used by ingress may fail to initialize due to pull rate limits. If this happens, and ingress fails to enable, you add a secret for the pulls for the following service accounts:
```

Comment:
Pull rate limit is appearing in both this and the minikube setup doc. We should have it at only place, no?